code/modules/flufftext/Hallucination.dm

```diff
@@ -391,11 +389,9 @@ Gunshots/explosions/opening doors/less rare audio (done)
 /obj/effect/fake_attacker/New(loc,var/mob/living/carbon/T)
 	..()
 	my_target = T
-	spawn(300)
-		qdel(src)
+	QDEL_IN(src, 300)
 	step_away(src,my_target,2)
-	spawn(0)
```

Comment:
...separate... thread? Replacing spawn(0) with addtimer(time=0) calls works and produces expected behaviour.